### PR TITLE
Hotfix to make the HcalCalPedestals run in 74X

### PR DESCRIPTION
**Pull request by mmusich**
This is needed to make the HCalCalPedestals run in the AlCaReco matrix for 50ns data.
Profiting of this I cleaned also `AlCaRecoOutput_cff.py` for multiple inclusions of AlcaReco DQM instances. 
This is due to non complete backport of #8350 within #9162. @kodolova @bsunanda @diguida you might want to follow this.

### CODE REVIEW

**Comment by diguida** [comment]
@mmusich thanks for the careful inspection!

**Comment by diguida** [comment]
please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/10054	05d53c5
- impact on reco is minimal ( EventFilter/HcalRawToDigi/plugins/HcalCalibTypeFilter.cc wouldn't pass events without raw data input)
- jenkins tests passed and comparisons with baseline show no difference

**Comment by slava77** [comment]
@mmusich please make a PR in 75X as well

**Comment by mmusich** [comment]
@slava77 the relevant commits were already submitted by @kodolova in #8350.

**Comment by slava77** [comment]
@mmusich thanks, good to know it's all taken care of in >=75X

**Comment by davidlange6** [comment]
Hi all
I realize a conflict in the original PR.

https://github.com/cms-sw/cmssw/blob/68b4ccfc9d6f7b0ae0980969622686957e61fc9a/Calibration/HcalAlCaRecoProducers/python/ALCARECOHcalCalPedestal_cff.py

presumably changes the settings in the NZS HCAL reco modules (instead of properly cloning modules and then changing the parameters of the clones). This is a problem because all AlCa configs get included if any AlCa is run in the job via

https://github.com/cms-sw/cmssw/blob/6979e4dd2afb50ec5ce7fac40c00d9e2dc8183f5/Configuration/StandardSequences/python/AlCaRecoStreams_cff.py

we need the modules to be properly cloned before they can be modified.

**Comment by mmusich** [comment]
@davidlange6 @kodolova I changed `Calibration/HcalAlCaRecoProducers/python/ALCARECOHcalCalPedestal_cff.py` to cope with your suggestion. Is this ok now? I also tested the WF through:
```
cmsDriver.py HCalCalPedestals -s ALCA:HcalCalPedestal --data --scenario=pp -n 100 --conditions auto:run2_data_FULL --datatier USER --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --customise=SLHCUpgradeSimulations/Configuration/postLS1Customs.customisePostLS1 --magField 38T_PostLS1 --processName=USER --filein root://eoscms//eos/cms/store/data/Commissioning2013/TestEnablesEcalHcalDT/RAW/v1/00000/12D70926-F147-E311-8EB4-003048F1B916.root --fileout file:RelVal_ALCA_HCalPedestals.root
```
in line with the tests from Olga (https://github.com/cms-sw/cmssw/pull/8350/files#diff-51267a96038f8382ac9fe87aabe07065R5) but I still see lots of:

```
Begin processing the 35th record. Run 216391, Event 1097, LumiSection 1 at 07-Jul-2015 14:40:30.736 CEST
%MSG-w L1GtTrigReport:  L1GtTrigReport:hltL1GtTrigReport  07-Jul-2015 14:40:30 CEST Run: 216391 Event: 1097

  L1GlobalTriggerReadoutRecord with input tag hltGtDigis not found.
  Event classified as Error
```

**Comment by Martin-Grunewald** [comment]
I suggest to remove
```
--customise=HLTrigger/Configuration/CustomConfigs.L1THLT
```
as this is only usefull if L1T/HLT is actually run the this cmsRun job.

File: Calibration/HcalAlCaRecoProducers/python/ALCARECOHcalCalPedestal_cff.py [hunk 1]
```diff
@@ -4,32 +4,68 @@
...
-horecoMB.digiLabel = 'hcalDigiAlCaMB'
+#from RecoLocalCalo.Configuration.hcalLocalRecoNZS_cff import *
+import RecoLocalCalo.HcalRecProducers.HcalSimpleReconstructor_hbhe_cfi
+hbherecoPedestals = RecoLocalCalo.HcalRecProducers.HcalSimpleReconstructor_hbhe_cfi.hbheprereco.clone()
 
+import RecoLocalCalo.HcalRecProducers.HcalSimpleReconstructor_hf_cfi
+hfrecoPedestals = RecoLocalCalo.HcalRecProducers.HcalSimpleReconstructor_hf_cfi.hfreco.clone()
+
+import RecoLocalCalo.HcalRecProducers.HcalSimpleReconstructor_ho_cfi
+horecoPedestals = RecoLocalCalo.HcalRecProducers.HcalSimpleReconstructor_ho_cfi.horeco.clone()
```
Comment (slava77) [inline Calibration/HcalAlCaRecoProducers/python/ALCARECOHcalCalPedestal_cff.py]: this looks like a redundant copy paste

Comment (mmusich) [inline Calibration/HcalAlCaRecoProducers/python/ALCARECOHcalCalPedestal_cff.py]: @slava77 thanks for spotting the spurious multiple imports. Should be clean now.

**Comment by mmusich** [comment]
@cmsbuild please aknowlege changes

**Comment by mmusich** [comment]
please test

**Comment by slava77** [comment]
@cmsbuild please test
hoping to pass this das error in the last test

**Comment by slava77** [comment]
@smuzaffar please check:
jenkins comparison with the baseline failed
https://cmssdt.cern.ch/jenkins/job/compare-root-files-short-matrix/8563/console

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/10054	54cfd3a
- code changes look fine for what concerns RECO code or module instances
- jenkins tests passed ok; the comparisons with the baseline is not available, unfortunately

**Comment by davidlange6** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
+1

**Comment by mmusich** [comment]
@davidlange6 maybe you can find some solution to the issue in https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/1885.html.
I'll need to forwardport it of course...

**Comment by davidlange6** [comment]
Hi Marco

yes, this is the needed fix. Too bad it wasn’t pushed forward in the first place. would have saved a bunch of work…

Thanks
David

**Comment by mmusich** [comment]
@davidlange6 sorry for that, I was waiting some feedback from Hcal DPG.

**Comment by bsunanda** [comment]
I pushed the changes needed in ALCARECOHcalCalPedestals by doing merge 10054 to CMSSW_7_4_X_2015-07-29-2300 and get from the push command HEAD -> fixHcalPedestals_76X. Shall I sen you the modified file?
________________________________

**Comment by diguida** [comment]
@bsunanda this is already done in #10478 Can you please check your changes w.r.t. that PR?

**Comment by bsunanda** [comment]
Yes - the changes are in the definition of the last sequence. 10478 has
seqALCARECOHcalCalPedestal = cms.Sequence(hcalCalibPedestal*
                                          hcalDigiAlCaPedestals*
                                          gtDigisAlCaPedestals*
                                          hcalLocalRecoSequencePedestals*
                                          hbherecoPedestals*
                                          hfrecoPedestals*
                                          horecoPedestals)
while there is double usage- so I drop the last 3 and keep
seqALCARECOHcalCalPedestal = cms.Sequence(hcalCalibPedestal*
                                          hcalDigiAlCaPedestals*
                                          gtDigisAlCaPedestals*
                                          hcalLocalRecoSequencePedestals)

________________________________

**Pull request merged**